targettests/execution/cudaq_run.cpp

```diff
@@ -0,0 +1,68 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2025 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+// RUN: nvq++ -fenable-cudaq-run %cpp_std %s -o %t && %t | FileCheck %s
+
+#include <cudaq.h>
+
+__qpu__ int nullary_test() {
+  unsigned result = 0;
+  cudaq::qvector v(8);
+  h(v);
+  z(v);
+  for (int i = 0; i < 8; i++) {
+    bool w = mz(v[i]);
+    result |= ((unsigned)w) << (8 - 1 - i);
+  }
+  return result;
+}
+
+__qpu__ int unary_test(int count) {
+  unsigned result = 0;
+  cudaq::qvector v(count);
+  h(v);
+  z(v);
+  for (int i = 0; i < count; i++) {
+    bool w = mz(v[i]);
+    result |= ((unsigned)w) << (count - 1 - i);
+  }
+  return result;
+}
+
+int main() {
+  int c = 0;
+  {
+    std::vector<int> results =
+        cudaq::run<int>(100, std::function<int()>{nullary_test});
```

Comment:
shouldn't the `<int>` in `run<int>` be unnecessary (inferred by the passed function return type)?

Reply:
Correct, It isn't required.